scriptworker/cot/verify.py

```diff
@@ -325,6 +325,8 @@ def guess_worker_impl(link):
         worker_impls.append("generic-worker")
     if task['payload'].get("osGroups") is not None:
         worker_impls.append("generic-worker")
+    if task['tags'].get("worker-implementation") is not None:
```

Comment:
We may want

`if task.get('tags', {}).get("worker-implementation") is not None:`

in case the task doesn't have a `tags` dict. If `tags` is required by the schema, we can keep this, but we'd need to update all the scriptworker test tasks to include `tags`.

Reply:
Ahhh, I hadn't thought about that breaking the tests. Sorry about that!

Indeed [tags is a required property](https://github.com/taskcluster/taskcluster-queue/blob/fd8027f1231421e95f11399a45aac3e59ade8525/schemas/v1/task.yml#L192) which is always present in queue responses, but I hadn't considered the tests.

I can push a new commit which updates the tests, or as you say we could switch to your proposed version, which would avoid needing to touch the tests and would still work if `tags` became optional at a point in the future. On reflection, I tend to prefer your proposed alternative if statement.